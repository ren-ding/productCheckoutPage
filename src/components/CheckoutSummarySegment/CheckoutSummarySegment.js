import React, {Component} from 'react';
import PropTypes from 'prop-types';

export default class CheckoutSummarySegment extends Component {
    render() {
        return (
            <div className='checkout-summary-wrapper'>
                {this.renderPromotionCode()}
                <div>Total Price: {this.props.total.toFixed(2)}  </div>
            </div>
        );
    }

    renderPromotionCode = () => {
        if(this.props.code)
            return (
                <div>Promotion Code: {this.props.code}  </div>
            )
    }
}

CheckoutSummarySegment.propTypes = {
    code: PropTypes.string.isRequired,
    total: PropTypes.number.isRequired
};