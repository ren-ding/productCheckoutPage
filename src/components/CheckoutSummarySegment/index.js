import CheckoutSummarySegment from './CheckoutSummarySegment';
export default CheckoutSummarySegment;