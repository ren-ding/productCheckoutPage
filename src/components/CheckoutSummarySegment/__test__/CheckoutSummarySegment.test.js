import React from 'react';
import CheckoutSummarySegment from '../CheckoutSummarySegment';
import renderer from 'react-test-renderer';

describe('CheckoutSummarySegment', ()=> {
    let props;

    beforeEach(()=> {
        props = {
            code: 'YYGWKJD',
            total: 200
        }   
    });

    describe('render', () => {
       it('should render to match snapshot', () => {
           const tree = renderer.create(<CheckoutSummarySegment {...props}/>).toJSON();
           expect(tree).toMatchSnapshot();
       }); 
    });

    
});