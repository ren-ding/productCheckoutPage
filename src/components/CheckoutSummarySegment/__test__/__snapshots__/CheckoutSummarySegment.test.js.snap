// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutSummarySegment render should render to match snapshot 1`] = `
<div
  className="checkout-summary-wrapper"
>
  <div>
    Promotion Code: 
    YYGWKJD
      
  </div>
  <div>
    Total Price: 
    200.00
      
  </div>
</div>
`;
